### ISSUE TITLE
Spike: Review use of long lists on TC

### ISSUE THREAD

**Issue by camerojo** [talentbeyondboundaries/tbbtalentv2#962]
As a developer
I would like to review the use of long lists on the TC
because...

- Understanding why long lists are being created and exported may point us to shortcomings that we can address in the TC which will remove the need for creating such lists
- The processing of long lists may consume resources that impact our performance

fyi @sadatmalik

**Comment by sadatmalik** [comment]
Adding a couple of thoughts for further consideration:

- Does it make sense to limit the maximum allowable size of a list to some reasonably large, but not too large, number
- Is there a need to add a CSV export function (alongside the Google sheet export)?

But as you say @camerojo we should first understand why long lists are being created and exported.

**Comment by camerojo** [comment]
I ran the following on prod to check large lists (2000+) and who was creating them...

select saved_list.id, saved_list.name, saved_list.created_date, users.first_name, users.last_name from saved_list join users on saved_list.created_by = users.id where (select count(*) count from candidate_saved_list where candidate_saved_list.saved_list_id = saved_list.id) > 1999

Here were the results

<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL3ZLQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--9ba90c0cfd59bad1bf24fc55c9a95a0825ecede1/Screenshot%202024-05-09%20at%205.54.30%E2%80%AFPM.png" alt="Screenshot 2024-05-09 at 5.54.30 PM.png" />

I think we need to understand why they are needed. Why not just define a search?
@sadatmalik

**Comment by samschlicht** [comment]
@camerojo just confirming I've deleted my large lists — they were used for the bulk transfer process I use to rescue an over-full CandidateData Google Shared Drive (process outline [here](https://app.tettra.co/teams/talentbeyondboundaries/pages/recurring-tc-processes#header-3saao-emergency-storage-measure)).

I can confirm that CPU spiked when I loaded them (to around 60%) so this is probably a worthwhile area to explore and perhaps add some user limits.

**Comment by samschlicht** [comment]
I would flag another type of user behaviour that I know is happening and causing spikes: 
spreadsheet exports of very large search results.

The reason one user gave to me for doing this is that they want to keyword within specific free-text fields such as candidate_education.course_name. If they're looking for someone with a nursing degree, using Keyword Search throws up lots of results where 'nursing' was entered in a different field. 

I've documented this as a possible new search feature in #825.

I tested exporting a sheet of around 25k candidates and I can confirm that this causes a massive spike!

<img width="988" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdzRTQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--69cd313c9d1f08b7dcb4fa4f5587b51f6bbee571/Screenshot%202024-05-27%20at%2011.19.53%E2%80%AFAM.png" alt="Screenshot 2024-05-27 at 11.19.53 AM.png" />

(This is a 10-minute spike that only ended with hard refresh of my candidate search.)

I would suggest placing a limit on spreadsheet exports and considering my feature suggestion. Also perhaps a code review for this function?

FYI @sadatmalik @camerojo
